src/cmake/build_PNG.cmake

```diff
@@ -0,0 +1,67 @@
+# Copyright Contributors to the OpenImageIO project.
+# SPDX-License-Identifier: Apache-2.0
+# https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+######################################################################
+# PNG by hand!
+######################################################################
+
+set_cache (PNG_BUILD_VERSION 1.6.44 "PNG version for local builds")
+set (PNG_GIT_REPOSITORY "https://github.com/glennrp/libpng")
+set (PNG_GIT_TAG "v${PNG_BUILD_VERSION}")
+
+set_cache (PNG_BUILD_SHARED_LIBS ${LOCAL_BUILD_SHARED_LIBS_DEFAULT}
+           DOC "Should execute a local PNG build, if necessary, build shared libraries" ADVANCED)
+
+string (MAKE_C_IDENTIFIER ${PNG_BUILD_VERSION} PNG_VERSION_IDENT)
+
+unset (PNG_FOUND)
+unset (PNG_LIBRARIES)
+unset (PNG_INCLUDE_DIRS)
+unset (PNG_INCLUDE_DIR)
+unset (PNG_VERSION_STRING)
+unset (PNG_DEFINITIONS)
+unset (PNG_VERSION)
+
+build_dependency_with_cmake (PNG
+    VERSION         ${PNG_BUILD_VERSION}
+    GIT_REPOSITORY  ${PNG_GIT_REPOSITORY}
+    GIT_TAG         ${PNG_GIT_TAG}
+    CMAKE_ARGS
+        -D PNG_SHARED=OFF
+        -D PNG_STATIC=ON
+        -D PNG_EXECUTABLES=OFF
+        -D PNG_TESTS=OFF
+        -D PNG_TOOLS=OFF
+        -D PNG_FRAMEWORK=OFF
+        -D CMAKE_POSITION_INDEPENDENT_CODE=ON
+        -D CMAKE_INSTALL_LIBDIR=lib
+        -D PNG_PREFIX=oiio
+    )
+
+    
+set (PNG_REFIND FALSE)
+set (PNG_REFIND_VERSION ${PNG_BUILD_VERSION})
+
+
+set (PNG_REFIND_ARGS EXACT REQUIRED)
+set (PNG_DIR ${PNG_LOCAL_INSTALL_DIR}/lib/cmake/PNG)
+set (PNG_FIND_VERSION_EXACT ON)
+
+if (PNG_BUILD_VERSION VERSION_GREATER 1.6.43)
+    list (APPEND PNG_REFIND_ARGS CONFIG)
+endif ()
+
+find_package(PNG ${PNG_REFIND_VERSION} ${PNG_REFIND_ARGS}
+             HINTS 
+                    ${PNG_LOCAL_INSTALL_DIR}/lib/cmake/PNG
+                    ${PNG_LOCAL_INSTALL_DIR}
+             NO_DEFAULT_PATH
```

Comment:
Have you found that NO_DEFAULT_PATH works as expected? I haven't been having much luck getting find_package to ignore system paths....

Reply:
I ended up with a collection of fixes that together worked. It's possible that some of them individually could be reverted, but I ran out of time or patience to continue to experiment with turning the all off one by one. I think this is certainly not harmful, and might be important.